build.mill

```diff
@@ -274,6 +274,8 @@ trait Chisel extends CrossSbtModule with HasScala2MacroAnno with HasScala2Plugin
 
   override def moduleDeps = super.moduleDeps ++ Seq(coreModule, svsimModule)
 
+  def compileIvyDeps = Agg(v.scalatest)
+
```

Comment:
Just to sanity check (because in my experience, publishing stuff is finicky and especially so since unipublish is a Jack Koenig special), can you check that the scalatest dependency is showing up properly as "provided" in the unipublish POM?

Reply:
There was a missing part to include this in the unipublish flow. This is fixed via: https://github.com/chipsalliance/chisel/pull/4669/commits/c8853914dc5d4afcf82ac4ac3378878f850d56c3

With that in place, scalatest is marked as "provided" scope in the POM. If you then try to fetch this with `cs fetch -e 2.13.15 org.chipsalliance::chisel:<your version>` then this will _not_ try to fetch scalatest.

To test the inverse, I changed the dependency from a `compileIvyDeps` to an `ivyDeps` and it _would_ try to fetch it. (I was not able to show the difference through manual POM edits.)